Feature/terraform implementation fix (#60)
Azure/aca-landing-zone-accelerator: scenarios/aca-internal/terraform/modules/04-container-apps-environment/README.md

## Expected results

The application platform, Azure Containers Apps, and its logging sinks within Azure Monitor will now be deployed. The workload is not deployed as part of step. Non-mission critical workload lifecycles are usually not tied to the lifecycle of the application platform, and as such are deployed isolated from infrastructure deployments, such as this one. Some cross-cutting concerns and platform feature enablement is usually handled however at this stage.

![A picture of the resources of this architecture, now with the application platform.](./media/container-apps-environment.png)

### Resources

- Container Apps Environment Environment
- Log Analytics Workspace
- Application Insights (optional)
- Dapr Telemetry with Application Insights (optional)
- Private DNS Zone for Container Apps Environment

## Steps

If you want to use remote storage, uncomment the backend block in the `providers.tf` file and provide the information for your Azure Storage Account. 

1. Create the Azure Container Apps application platform resources.

```bash
terraform init
terraform plan -out tfplan

> terraform apply tfplan 

Reviewer: Same, no `cd` no change.

- Dapr Telemetry with Application Insights (optional)
- Private DNS Zone for Container Apps Environment

#### Configure Terraform remote state

To configure your Terraform deployment to use the newly provisioned storage account and container, edit the [`./providers.tf`](./providers.tf) file at lines 11-13 as below:

```hcl

>   backend "azurerm" {
>     resource_group_name  = "<REPLACE with $RESOURCE_GROUP_NAME>"
>     storage_account_name = "<REPLACE with $STORAGE_ACCOUNT_NAME>"
>     container_name       = "tfstate"
>     key                  = "myapp/container-apps-environment.tfstate"
>   }

Reviewer: ```suggestion
backend "azurerm" {
  resource_group_name  = "<REPLACE with $RESOURCE_GROUP_NAME>"
  storage_account_name = "<REPLACE with $STORAGE_ACCOUNT_NAME>"
  container_name       = "tfstate"
  key                  = "myapp/container-apps-environment.tfstate"
}
```

```

* `resource_group_name`: Name of the Azure Resource Group that the storage account resides in.
* `storage_account_name`: Name of the Azure Storage Account to be used to hold remote state.
* `container_name`: Name of the Azure Storage Account Blob Container to store remote state.
* `key`: Path and filename for the remote state file to be placed in the Storage Account Container. If the state file does not exist in this path, Terraform will automatically generate one for you.

## Steps

> 1. Navigate to the Terraform module for the hub. 

Reviewer: ```suggestion
1. Navigate to the Terraform module for the ACA resources.
```

   
   ```bash

>    cd modules/04-container-apps-environment

Reviewer: ```suggestion
   cd ../04-container-apps-environment
```
